### PR TITLE
Improve canConnectToExternal()

### PR DESCRIPTION
**Pull request by Pilzinsel64**
Fixes a case where at loading the Grid sometimes the conduits may not connect.
I've tested this case with all conduit types and all me cables and some buses and mashines, still work as it should.

Before:
grafik

After:
grafik

### CODE REVIEW

File: src/main/java/crazypants/enderio/conduit/me/MEConduit.java [hunk 1]
```diff
@@ -174,6 +174,9 @@ public boolean canConnectToExternal(ForgeDirection dir, boolean ignoreDisabled)
                 if (node == null) {
                     node = part.getGridNode();
                 }
+                if (node == null) {
```
[the remark this answers is not in the document]
Comment (Pilzinsel64) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: Not sure. But with my Ultra Dense PR I had to change the code of this method. However, my change with this PR here adds a missing case from the code BEFORE my Ultra Dense PR to cover a few cases where ME Conduits wont connect anymore now (one example shown on the picture).

[the remark this answers is not in the document]
Comment (Pilzinsel64) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: @repo-alt See the previous code here: https://github.com/GTNewHorizons/EnderIO/blob/b22915cc9cc2903fed7f3d9cb4936025b30df644/src/main/java/crazypants/enderio/conduit/me/MEConduit.java#LL155C37-L155C37
It doesn't even check the node itself.

[the remark this answers is not in the document]
Comment (repo-alt) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: I don't quite understand, sorry

[the remark this answers is not in the document]
Comment (Pilzinsel64) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: @repo-alt I'm sorry, let me try again...

Before my PR that adds Ultra Dense Me Conduits the code was like this:
https://github.com/GTNewHorizons/EnderIO/blob/b22915cc9cc2903fed7f3d9cb4936025b30df644/src/main/java/crazypants/enderio/conduit/me/MEConduit.java#LL155C37-L155C37

Then I had to adjust the that code for canConnectToExternal() to get a similar behavior as the AE2 cables.

But this adjusted method doesn't catch the case anymore that the GridNode can be null, so it won't connect to AE2 cables in a few cases (one shown in the picture at the top).

Now, with this PR I want to fix that by handling this simple case where there is no GridNode.

[the remark this answers is not in the document]
Comment (repo-alt) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: Yes, I understand that, I just asked if it is possible to have a part w/o a grid node (how is that even possible?) facing the conduit. Obviously it should not connect in that case, and that was checked before.

[the remark this answers is not in the document]
Comment (Pilzinsel64) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: Hm, the only thing we could do here is to provide a method like `canConnectFrom(direction)` via the AE2 API and shrink the code of this method `canConnectFrom()` here probably to a few lines.

But I suggest to do that in another step and I suggest to merge this as first step as otherwise some ME Systems could get broken softly.

[the remark this answers is not in the document]
Comment (repo-alt) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: You see, the code you add only works in the situation where there is a part attached to the cable, but in the picture you provide there is no part, so, how exactly can that change affect the situation on the picture?
Also, maybe there is an easy way to reproduce it? If so I might have a look what's going on there

[the remark this answers is not in the document]
Comment (Pilzinsel64) [reply src/main/java/crazypants/enderio/conduit/me/MEConduit.java:177]: Ah, now I understand, I think...
There is a part on the picture: PartGlassCable. This is what the debugger says.

To reproduce:
1. Put three me interfaces and connect them via a fluix glass cable. Then put a another glass cable below, as shown in the picture. Then, under this cable now put a me conduit (normal or dense) and see it connects.
2. Save & Quit
3. Load world again
4. See the conduit doesn't connect unless you trigger a reconnection (e.g. with the wrench or by placing it again)

If you want I can provide you the test world shown on the picture.

**Comment by repo-alt** [comment]
That's what will happen if the change is implemented as is 
image
at least you should check if there is no directional part in that direction.
I don't like that at all, but I don't see an easy solution, because EIO and AE use different initialization modes and grid nodes may be either present or not at the time of the conduit initialzation. Better solution would be to have a way to reset `AbstractConduit.connectionsDirty` in case when the derived class encounters "neighbours are not quite loaded" situation.

**Comment by Pilzinsel64** [comment]
Ah, I see. Hm, not sure what a "directional part" is, but you second idea sounds good.
Check my latest commit `needUpdateConnections`, is that what you mean? @repo-alt

Sure it's not a good way how I implemented it at all yet, but it seems to work. What do you think?
grafik

**Review by repo-alt — APPROVED** [review]

**Comment by repo-alt** [comment]
That is at least more universal solution
